_src-lp/blocks/b-productswithselectors/__TESTS__/b-productswithselectors.spec.js

```diff
@@ -0,0 +1,67 @@
+// eslint-disable-next-line import/extensions
+import mockData from './mock-data.json';
+// mock everything that is not related to the actual main scope
+jest.mock('../../../scripts/lib-franklin.js', () => ({
+  ...jest.requireActual('../../../scripts/lib-franklin.js'),
+  loadHeader: jest.fn(),
+  loadFooter: jest.fn(),
+}));
+
+describe('b-productswithselectors block', () => {
+
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+  });
+
+  describe('b-productswithselectors', () => {
+
+    // init / bootstrap page()
+    document.body.innerHTML = mockData.b_productswithselectors;
+    import('../../../scripts/scripts.js');
+
+    let b_productswithselectors;
+
+    beforeEach(() => {
+      b_productswithselectors = document.querySelector('.b-productswithselectors-container');
+    });
+
+    test('check if products have been added to the window.productsListCount', () => {
+        expect(window.productsListCount).toBeTruthy();
+    });
+
+    test('check if the number of servers and mailboxes changes based on number of devices selected', () => {
+      const usersSelectorBox = b_productswithselectors.querySelectorAll('.selectorBox')[0];
+    
+      // Set the target value on the selector
+      usersSelectorBox.value = '11';
+      // Dispatch the change event
+      usersSelectorBox.dispatchEvent(new Event('change'));
+
+      const triggerValue = usersSelectorBox.value;
+      const fileServers1stProd = 4;
+      const fileServers2ndProd = 4;
+      const mailboxes = 17;
+
+      const selectors = [
```

Comment:
naming is to general; which selectors ? what is index, what is type ? maybe add more verbosity in your code otherwise seems hard to follow the logic you have written; please assert also the current and discounted prices

Reply:
updated data is not uniform, 3 boxes need to be tested, as to code verbosity, why do i need to be more verbose, since I test all the selectors ? I don't test prices in this logic anyway

Reply:
"I don't test prices in this logic anyway": the price is changing also after the initial selection, so we need to test that also
"as to code verbosity, why do i need to be more verbose, since I test all the selectors ?": because those "indexes" and listChildNumber are way to general, actually you are referring to: discountedPrice, currentPrice, desktopAndLaptops, fileServers, mailboxes. The test needs to be clear what what needs to be tested / asserted